test/integration/stdio_transport_test.go

```diff
@@ -0,0 +1,571 @@
+package integration
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// StdioTestServer represents a server instance for stdio transport testing
+type StdioTestServer struct {
```

Comment:
This type seems to be functionally equivalent to the `ComprehensiveTestServer`, can we re-use that? Also same question about using the mcp-go client instead of recreating all of that logic here again